### PR TITLE
Enable custom attributes for events

### PR DESCRIPTION
**Pull request by alejandrodnm**
#### Description of the changes

This PR allows events with attributes to be decorated with the entity `customAttributes` when they are added to via the `entity.AddEvent` function.

### Author

- [x ] add a risk label after carefully considering the "blast radius" of your changes
- [x ] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x ] assign at least one reviewer
- [x ] document feature

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] request documenation about anything that isn't clear and obvious
- [ ] document agent version supporting the feature

### CODE REVIEW

File: data/attributes/attributes.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package attributes
```
Comment (varas) [inline data/attributes/attributes.go]: singular for package names please

Comment (varas) [reply data/attributes/attributes.go]: is there any need to wrap this `Attribute` type into its own `attribute` package?

IMHO it feels more dev friendly to remain at the `metric` one for:
- atomicity: you won't use attributes without metrics
- simplicity: metrics package gets easier to use
- readability/usage: you don't need to type package name all around

File: integration/entity_test.go [hunk 2]
```diff
@@ -103,6 +104,29 @@ func TestAddNotificationEvent(t *testing.T) {
 	}
 }
 
+func TestAddEventWithAttributes(t *testing.T) {
+	en, err := newEntity("Entity1", "Type1", persist.NewInMemoryStore(), false)
+	if err != nil {
+		t.Fatal(err)
```
Comment (varas) [inline integration/entity_test.go]: require.NoError

File: integration/entity_test.go [hunk 3]
```diff
@@ -103,6 +104,29 @@ func TestAddNotificationEvent(t *testing.T) {
...
+	}
+
+	en.customAttributes = attributes.Attributes{attributes.Attr("clusterName", "my-cluster-name")}
+	attrs := map[string]interface{}{"attrKey": "attrVal"}
+	err = en.AddEvent(event.NewWithAttributes("TestSummary", "TestCategory", attrs))
+	assert.NoError(t, err)
+
+	assert.Len(t, en.Events, 1)
```
Comment (varas) [inline integration/entity_test.go]: require.Len

File: data/metric/metrics.go [hunk 4]
```diff
@@ -59,20 +52,12 @@ func NewSet(eventType string, storer persist.Storer, attributes ...Attribute) (s
 }
 
 // AddCustomAttributes add customAttributes to MetricSet
-func AddCustomAttributes(metricSet *Set, customAttributes []Attribute) {
+func AddCustomAttributes(metricSet *Set, customAttributes []attributes.Attribute) {
```
Comment (varas) [inline data/metric/metrics.go]: out of scope:
bad initial design, this should have been method and variadic, now modifying will cause BC

**Comment by mariomac** [comment]
LGTM as long as Juan's comments are addressed

File: data/attribute/attribute.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package attribute
+
```
Comment (varas) [inline data/attribute/attribute.go]: https://github.com/newrelic/infra-integrations-sdk/pull/187#discussion_r326013176

File: data/attributes/attributes.go (hunk 1 above)
Comment (alejandrodnm) [reply data/attributes/attributes.go]: Sorry I missed this. The reason to abstract it away into its own package is that now it used in both in the `metric` and `event` package. If you want I can keep it all on the `metric` but then the `event` package will have to import `metric` to use the attributes.

**Comment by alejandrodnm** [comment]
@varas sorry i missed one of your comments, I replied here https://github.com/newrelic/infra-integrations-sdk/pull/187#discussion_r350369330

**Pull request merged**